libcxx/include/__config

```diff
@@ -1215,6 +1215,24 @@ typedef __char32_t char32_t;
 #    define _LIBCPP_HAS_EXPLICIT_THIS_PARAMETER 0
 #  endif
 
+// TODO(LLVM 22): Remove this macro once LLVM19 support ends.
+#  if __has_builtin(__reference_constructs_from_temporary) ||                                                          \
+      (defined(_LIBCPP_CLANG_VER) &&                                                                                   \
+       ((!defined(__ANDROID__) && _LIBCPP_CLANG_VER >= 1901) || (defined(__ANDROID__) && _LIBCPP_CLANG_VER >= 1900)))
+#    define _LIBCPP_HAS_REFERENCE_CONSTRUCTS_FROM_TEMPORARY 1
+#  else
+#    define _LIBCPP_HAS_REFERENCE_CONSTRUCTS_FROM_TEMPORARY 0
+#  endif
```

Comment:
This should just be `__has_builtin(__reference_constructs_from_temporary)`. 99% of people use the same version of Clang and libc++, and this is way too complicated of a condition compared to what we gain.